### PR TITLE
[edpm_ssh_known_hosts]Support non standard ssh port

### PR DESCRIPTION
**Pull request by gibizer**

### CODE REVIEW

**Review by rebtoor — APPROVED** [review]

**Comment by rebtoor** [comment]
recheck

retry_limit

File: roles/edpm_ssh_known_hosts/tasks/main.yml [hunk 1]
```diff
@@ -73,6 +73,13 @@
           {%       set _ = entries.append(hostdata['canonical_hostname']) %}
           {%     endif %}
           {%     set _ = entries.append(host ~ '*') %}
+          {%     if 'ansible_port' in hostdata and hostdata['ansible_port'] != 22 %}
```
Comment (gibizer) [inline roles/edpm_ssh_known_hosts/tasks/main.yml:76]: this needs to be `hostdata['ansible_port']|int` to get make the int comparison work

**Review by rebtoor — APPROVED** [review]

**Comment by gibizer** [comment]
recheck
```
2023-10-11 15:34:38.164 5 INFO config_tempest.constants [-] Uploading image 'cirros-0.5.2-x86_64-disk.img' from '/var/lib/tempest/openshift/etc/cirros-0.5.2-x86_64-disk.img'[00m
2023-10-11 15:34:38.166 5 CRITICAL tempest [-] Unhandled error: FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/tempest/openshift/etc/cirros-0.5.2-x86_64-disk.img'
2023-10-11 15:34:38.166 5 ERROR tempest Traceback (most recent call last):
2023-10-11 15:34:38.166 5 ERROR tempest   File "/usr/bin/discover-tempest-config", line 10, in <module>
2023-10-11 15:34:38.166 5 ERROR tempest     sys.exit(main())
2023-10-11 15:34:38.166 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/main.py", line 597, in main
2023-10-11 15:34:38.166 5 ERROR tempest     config_tempest(
2023-10-11 15:34:38.166 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/main.py", line 557, in config_tempest
2023-10-11 15:34:38.166 5 ERROR tempest     image.create_tempest_images(conf, retry_alt=retry_alt)
2023-10-11 15:34:38.166 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/services/image.py", line 137, in create_tempest_images
2023-10-11 15:34:38.166 5 ERROR tempest     image_id = self.find_or_upload_image(image_id, name,
2023-10-11 15:34:38.166 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/services/image.py", line 197, in find_or_upload_image
2023-10-11 15:34:38.166 5 ERROR tempest     image = self._upload_image(image_name, image_dest)
2023-10-11 15:34:38.166 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/services/image.py", line 246, in _upload_image
2023-10-11 15:34:38.166 5 ERROR tempest     with open(path, 'rb') as data:
2023-10-11 15:34:38.166 5 ERROR tempest FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/tempest/openshift/etc/cirros-0.5.2-x86_64-disk.img'
2023-10-11 15:34:38.166 5 ERROR tempest [00m
{0} setUpClass (tempest.scenario.test_network_basic_ops.TestNetworkBasicOps) ... SKIPPED: Neutron not available
```

**Comment by gibizer** [comment]
recheck
```
2023-10-11 17:56:33.137 5 INFO config_tempest.constants [-] Uploading image 'cirros-0.5.2-x86_64-disk.img' from '/var/lib/tempest/openshift/etc/cirros-0.5.2-x86_64-disk.img'[00m
2023-10-11 17:56:33.139 5 CRITICAL tempest [-] Unhandled error: FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/tempest/openshift/etc/cirros-0.5.2-x86_64-disk.img'
2023-10-11 17:56:33.139 5 ERROR tempest Traceback (most recent call last):
2023-10-11 17:56:33.139 5 ERROR tempest   File "/usr/bin/discover-tempest-config", line 10, in <module>
2023-10-11 17:56:33.139 5 ERROR tempest     sys.exit(main())
2023-10-11 17:56:33.139 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/main.py", line 597, in main
2023-10-11 17:56:33.139 5 ERROR tempest     config_tempest(
2023-10-11 17:56:33.139 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/main.py", line 557, in config_tempest
2023-10-11 17:56:33.139 5 ERROR tempest     image.create_tempest_images(conf, retry_alt=retry_alt)
2023-10-11 17:56:33.139 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/services/image.py", line 137, in create_tempest_images
2023-10-11 17:56:33.139 5 ERROR tempest     image_id = self.find_or_upload_image(image_id, name,
2023-10-11 17:56:33.139 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/services/image.py", line 197, in find_or_upload_image
2023-10-11 17:56:33.139 5 ERROR tempest     image = self._upload_image(image_name, image_dest)
2023-10-11 17:56:33.139 5 ERROR tempest   File "/usr/lib/python3.9/site-packages/config_tempest/services/image.py", line 246, in _upload_image
2023-10-11 17:56:33.139 5 ERROR tempest     with open(path, 'rb') as data:
2023-10-11 17:56:33.139 5 ERROR tempest FileNotFoundError: [Errno 2] No such file or directory: '/var/lib/tempest/openshift/etc/cirros-0.5.2-x86_64-disk.img'
2023-10-11 17:56:33.139 5 ERROR tempest [00m
```

**Pull request merged**